kotlin/internal/opts.bzl

```diff
@@ -54,6 +61,13 @@ kt_kotlinc_options = rule(
 )
 
 _JOPTS = {
+    "all_warnings_as_errors": struct(
```

Comment:
`w_error`?

Reply:
I would update "warn" to be a trinary: "off", "report", "error", eg:

```
    "warn": struct(
        args = dict(
            default = "report",
            doc = "Control warning behaviour.",
            values = ["off", "report", "error"],
        ),
        type = attr.string,
    ),
```

Reply:
That's a really good idea, I'll update this later today.

Reply:
it seems like we're barreling towards reproducing all compiler flags and their options in this dictionary. Are we sure it doesn't make sense to allow arbitrary flags instead? Seeing a few more PRs open adding more of these as well

Reply:
The dictionary is merely a code reuse tool. The flags are typesafe, and as suggested above, can provide a richer api.

Having maintained and worked on rules that allow arbitrary string lists (or worse, just a string) the long term maintenance outweighs the expediency. There are plans to restructure the compiler calls to use the embeddable api, which means reparsing them in the worker. Finally, removal of 

From a code repository maintenance perspective, abstracting the flags allows:
* Defining flagset options and constraints (visibility, environment, etc.) them (makes large scale code changes a lot easier)
* Feedback for incorrect values happens in analysis time, as well as improved discovery as the compiler options are embedded with the rules.
* Auditing flag usage.
* Deprecations are immediately clear via warnings from the rules.

The initial commit was address the need to add flags in a  maintainable manner. I expect there will be a lot of prs adding to the options for a bit as the most used options are added.

Reply:
Thanks for all the details! I updated this to use a trinary, for both `javac` and `kotlinc`.